crates/ruff_python_formatter/src/lib.rs

```diff
@@ -176,6 +183,80 @@ mod tests {
         );
     }
 
+    #[test]
+    fn string_processing() {
+        use ruff_formatter::prelude::*;
+        use ruff_formatter::{format, format_args, write};
+
+        // 77 after g group (leading quote)
+        let fits =
+            r#"aaaaaaaaaa bbbbbbbbbb cccccccccc dddddddddd eeeeeeeeee ffffffffff gggggggggg h"#;
+        let breaks =
+            r#"aaaaaaaaaa bbbbbbbbbb cccccccccc dddddddddd eeeeeeeeee ffffffffff gggggggggg hh"#;
+
+        struct FormatString<'a>(&'a str);
+
+        impl Format<SimpleFormatContext> for FormatString<'_> {
+            fn fmt(
+                &self,
+                f: &mut ruff_formatter::formatter::Formatter<SimpleFormatContext>,
+            ) -> FormatResult<()> {
+                let format_str = format_with(|f| {
+                    write!(f, [text("\"")])?;
+
+                    let mut words = self.0.split_whitespace().peekable();
+                    let mut fill = f.fill();
+
+                    let separator = format_with(|f| {
+                        group(&format_args![
+                            if_group_breaks(&text("\"")),
+                            soft_line_break_or_space(),
```

Comment:
Correct. I fixed the whitespace. I won't implement the logic to keep multiple whitespace because the goal of this PR is only to verify that the IR works.